crates/core/tedge_mapper/src/core/component.rs

```diff
@@ -4,4 +4,6 @@ use tedge_config::TEdgeConfig;
 #[async_trait]
 pub trait TEdgeComponent {
     async fn start(&self, tedge_config: TEdgeConfig) -> Result<(), anyhow::Error>;
+    async fn init(&self) -> Result<(), anyhow::Error>;
+    async fn clear_session(&self) -> Result<(), anyhow::Error>;
 }
```

Comment:
By adding a `session_name()` method to this trait, you can implement a default `clear_session()` implementation and also move here the implementation of `get_mqtt_config()` that is duplicated 3 times.

```suggestion
pub trait TEdgeComponent {
    fn session_name(&self) -> &str;
    async fn start(&self, tedge_config: TEdgeConfig) -> Result<(), anyhow::Error>;
    async fn init(&self) -> Result<(), anyhow::Error>;
    
    async fn init_session(&self) -> Result<(), anyhow::Error> {
        // Use `self.mqtt_config()` to init the session
    }
    
    async fn clear_session(&self) -> Result<(), anyhow::Error> {
          // Use `self.mqtt_config()` to clear the session
    }
    
    fn get_mqtt_config() -> Result<mqtt_channel::Config, anyhow::Error> {
           // Use `self.session_mane()` to create the mqtt_config
     }
}
```

Reply:
fixed